solawi/api.py

```diff
@@ -233,42 +233,52 @@ def delete_bet(share_id: int, bet_id: int):
     return jsonify(), 204
 
 
-class SharePatchSchema(BaseModel, extra=Extra.forbid):
+class SharePatchSchema(BaseModel):
     note: Optional[str]
     archived: Optional[bool]
+    name: str | None
+
+    class Config:
+        orm_mode = True
+        extra = Extra.forbid
+
+
+class ShareSchema(BaseModel):
+    id: str
+    station_id: int
+    note: Optional[str]
+    archived: Optional[bool]
+    name: str | None = Field(alias="displayed_name")
+
+    class Config:
+        orm_mode = True
+        allow_population_by_field_name = True
 
 
 @api.patch("/shares/<int:share_id>")
 @login_required()
 @validate()
-def patch_share(body: SharePatchSchema, share_id: int):
+def patch_share(body: SharePatchSchema, share_id: int) -> ShareSchema:
     share = Share.get(share_id)
     for key, value in body.dict().items():
         setattr(share, key, value)
     share.save()
-    resp = share.json
-    return jsonify(share=resp)
-
-
-class ShareSchema(BaseModel):
-    id: str
-    station_id: int
-    note: Optional[str]
-    archived: Optional[bool]
+    resp = ShareSchema.from_orm(share)
+    return resp
 
 
 @api.route("/shares/<int:share_id>", methods=["POST"])
 @login_required()
 @validate()
-def post_shares_details(body: ShareSchema, share_id: int):
+def post_shares_details(body: ShareSchema, share_id: int) -> ShareSchema:
     share = Share.get(share_id)
     json = body.dict()
-    for field in ["station_id", "note", "archived"]:
```

Comment:
I think I had it like this so that it would not be possible for API users to change the `id` of a `share`. I think with the new code, if an attacker was to send the `id` field in the JSON payload it would also update that in the db, no? Maybe we can add a test to be sure?

Reply:
The list of field is controlled by the type we declare in the signature as `body`. here it's `SharePatchSchema`, that only have 
`note`, `archived` and `name` field. It's guaranteed by pydantic API that no other field could make it through

Reply:
But in this case, the type for `body` is actually `ShareSchema`, not `SharePatchSchema`, right? Reading this, I think that we probably want to get rid of the whole `post_shares_details` method and use `patch_share` for all updates to existing shares. That makes the whole thing more RESTful and is a rather minor change. Would you be fine with reverting your changes to this method in this PR and adding a `TODO` to replace it completely instead?

Reply:
You are right, my bad, I thought I read `SharePatchSchema`.
I will fix it, no problem.
Regarding the topic of removing `post_shares_details` I don't understand your suggestion yet, but I will take the time to think about it.